src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -123,6 +126,50 @@ uct_cuda_ipc_iface_flush(uct_iface_h tl_iface, unsigned flags,
     return UCS_INPROGRESS;
 }
 
+static ucs_status_t uct_cuda_ipc_iface_event_fd_get(uct_iface_h tl_iface, int *fd_p)
+{
+    uct_cuda_ipc_iface_t *iface = ucs_derived_of(tl_iface, uct_cuda_ipc_iface_t);
+
+    *fd_p = iface->eventfd;
+    return UCS_OK;
+}
+
+static void uct_cuda_ipc_common_cb(void *cuda_ipc_iface)
+{
+    uct_cuda_ipc_iface_t *iface = cuda_ipc_iface;
+    uint64_t dummy = 1;
+    int ret;
+
+    /* No error handling yet */
+    do {
+        ret = write(iface->eventfd, &dummy, sizeof(dummy));
+        if (ret == sizeof(dummy)) {
+            return;
+        } else if (ret == -1) {
+            if (errno == EAGAIN) {
+                continue;
+            } else if (errno != EINTR) {
+                ucs_error("Signaling wakeup failed: %m");
+                return;
+            }
+        } else {
+            ucs_assert(ret == 0);
+        }
+    } while (ret == 0);
+}
+
+#if (__CUDACC_VER_MAJOR__ >= 100000)
+static void CUDA_CB myHostFn(void *iface)
+{
+    uct_cuda_ipc_common_cb(iface);
+}
+#else
+void CUDA_CB myHostCallback(CUstream hStream,  CUresult status, void *iface)
```

Comment:
should it be static as well?
also, seems like only function header could be under `#if`/`#else`. and function body is the same in both cases